webhooks/finished/public/js/index.js

```diff
@@ -0,0 +1,110 @@
+let accountNames = {};
+
+const checkConnectedStatus = async function () {
+  const connectedData = await callMyServer("/server/get_user_info");
+  if (connectedData.user_status === "connected") {
+    document.querySelector("#connectedUI").classList.remove("invisible");
+  } else {
+    document.querySelector("#disconnectedUI").classList.remove("invisible");
+  }
+};
+
+const getBalances = async function () {
+  const balanceData = await callMyServer("/server/balances");
+  const simplifiedData = balanceData.accounts.map((acct) => {
+    return {
+      account: acct.name ?? "(Unknown)",
+      // Yes, making the incorrect assumption these are always in dollars.
+      account_mask: acct.mask ?? "",
+      current_balance:
+        acct.balances.current != null
+          ? `$${acct.balances.current.toFixed(2)}`
+          : "(Unknown)",
+    };
+  });
+  console.table(simplifiedData);
+  displaySimplifiedData(simplifiedData);
+};
+
+const createAssetReport = async function () {
+  const reportResponse = await callMyServer("/server/create_asset_report");
+  displaySimplifiedData([{ asset_report_id: reportResponse.asset_report_id }]);
+};
+
+const getTransactions = async function () {
+  const transactionData = await callMyServer("/server/transactions");
+  const simplifiedData = transactionData.transactions.map((t) => {
+    return {
+      date: t.date,
+      vendor: t.name,
+      category: t.category[0],
+      amount: `$${t.amount.toFixed(2)}`,
+    };
+  });
+  console.table(simplifiedData);
+  displaySimplifiedData(simplifiedData);
+};
+
+function displaySimplifiedData(simplifiedData) {
+  const output = document.querySelector("#output");
+  output.innerHTML = "<ul></ul>";
+  simplifiedData.forEach((thingToShow) => {
+    const nextItem = document.createElement("li");
+    nextItem.textContent = JSON.stringify(thingToShow);
+    output.appendChild(nextItem);
+  });
+}
+
+const fireTestWebhook = async function () {
+  await callMyServer("/server/fire_test_webhook", true);
```

Comment:
Could possibly build in options to send different webhooks, but that may be beyond scope of this app...

Reply:
Yeah. Also, the sandbox/item/fire_webhook endpoint only lets you send three webhooks. And one of them is for Auth (which we're not using) and the other is Transactions DEFAULT_UPDATE (which we'll probably be using less now that the Sync API is out).